src/hooks/system_mod.rs

```diff
@@ -0,0 +1,72 @@
+use rhai::{Dynamic, EvalAltResult, Module};
+use std::process::Command;
+
+use crate::{
+    interactive::prompt_for_variable,
+    project_variables::{StringEntry, TemplateSlots, VarInfo},
+};
+
+type Result<T> = anyhow::Result<T, Box<EvalAltResult>>;
+
+/// Creates the system module, containing the `command` function,
+/// which allows you to run system command.
+pub fn create_module(allow_commands: bool, silent: bool) -> Module {
+    let mut module = Module::new();
+
+    module.set_native_fn("command", move |name: &str, commands_args: rhai::Array| {
+        run_command(name, commands_args, allow_commands, silent)
+    });
+
+    module
+}
+
+fn run_command(
+    name: &str,
+    args: rhai::Array,
+    allow_commands: bool,
+    silent: bool,
+) -> Result<Dynamic> {
+    let args: Vec<String> = args.into_iter().map(|arg| arg.to_string()).collect();
+
+    // If --allow-commands is false, we need to prompt. But we shouldn't if we're in silent mode.
+    if !allow_commands && silent {
+        return Err("Cannot prompt for system command confirmation in silent mode. Use --allow-commands if you want to allow the template to run system commands in silent mode.".into());
```

Comment:
FYI: anyhow has a `bail` macro for early exit with an error

Reply:
This led me to realize that the Result type defined here overrides both type parameters of anyhow::Result, making it exactly the same as std::result::Result. So `bail` actually can't be used here and there is no reason to even use anyhow in this case. I unified the type definitions in a new `HookResult` type defined in the super-module that all the hook modules can use.